src/components/MapLoadService.js

```diff
@@ -0,0 +1,136 @@
+(function() {
+
+  var module = angular.module('ga_mapload_service', [
+  ]);
+
+ /**
+   * This service can be used to inspect loading of the map.
+   * It times the loading of the map and emits messages about
+   * it
+   */
+  module.provider('gaMapload', function() {
+
+    this.$get = function($window, gaLayerFilters, gaPermalink) {
+
+      var layersPending = 0;
+      var layersDone = 0;
+      var mapStartTime;
+
+      var updateMapLoading = function() {
+        if (layersPending == layersDone) {
+          var seconds = (new Date() - mapStartTime) / 1000;
+          mapStartTime = undefined;
+          $window.console.info('Map loaded in ' + seconds + 's');
+        }
+      };
+
+      var addLayerPending = function() {
+        if (mapStartTime == undefined) {
+          $window.console.info('Start loading map.');
+          mapStartTime = new Date();
+        }
+        layersPending++;
+        updateMapLoading();
+      };
+
+      var addLayerLoaded = function() {
+        layersDone++;
+        updateMapLoading();
+      };
+
+      var LayerProgress = function(layer) {
+        var src, startTime;
+        var loading = 0;
+        var loaded = 0;
+
+        var update = function() {
+          if (loading == loaded) {
+            var seconds = (new Date() - startTime) / 1000;
+            startTime = undefined;
+            $window.console.log('Layer ' + layer.id +
+                                ' finished loading in ' + seconds + 's.');
+            addLayerLoaded();
+          }
+        };
+
+        var addLoading = function() {
+          if (startTime == undefined) {
+            startTime = new Date();
+            addLayerPending();
+          }
+          loading++;
+          update();
+        };
+        var addLoaded = function() {
+          loaded++;
+          update();
+        };
+
+        if (layer instanceof ol.layer.Tile ||
+            layer instanceof ol.layer.Image) {
+          var start, error, end;
+          src = layer.getSource();
+          if (src instanceof ol.source.WMTS ||
+              src instanceof ol.source.TileWMS) {
+              start = 'tileloadstart';
+              error = 'tileloaderror';
+              end = 'tileloadend';
+          } else if (src instanceof ol.source.ImageWMS) {
+              start = 'imageloadstart';
+              error = 'imageloaderror';
+              end = 'imageloadend';
+          }
+          if (start) {
+            src.on(start, function() {
+              addLoading();
+            });
+            src.on(error, function() {
+              addLoaded();
+            });
+            src.on(end, function() {
+              addLoaded();
+            });
+          }
+        }
+      };
+
+      var progs = {};
+
+      var Mapload = function() {
+        this.init = function(scope) {
+          // Only active if debug=true is specified
+          if (gaPermalink.getParams().debug != 'true') {
+            return;
+          }
```

Comment:
For now, I agree. I think this service could be used in non-debug version as well. It might be good to know for some functionality (like when trying the get an image from the map, see https://github.com/geoadmin/mf-geoadmin3/pull/3464) when the map has finished loading.

But we can then switch it back when needed.

Good review, I'll adapt.